crates/parser/src/format/character_separated/detection.rs

```diff
@@ -106,9 +119,14 @@ pub fn detect_dialect(
         .pop()
         .expect("must have at least one candidate dialect");
     // Log the top few candidates, as it's helpful to see the runner up when detection doesn't go as we expected
-    let runners_up = &dialects[0..(dialects.len().min(3))];
+    let runners_up = &dialects[dialects.len().saturating_sub(3)..];
```

Comment:
The previous expression here was incorrect. So this now correctly prints the top 3 runners up, with the last value being the 2nd place finisher.